back/engines/commercial/analytics/db/migrate/20220624100104_create_dimension_dates.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class CreateDimensionDates < ActiveRecord::Migration[6.1]
+  def up
+    create_table :analytics_dimension_dates, id: false do |t|
+      t.date :date, primary_key: true
+      t.string :year
+      t.string :month
+      t.string :day
+    end
+  end
+
+  def down
+    drop_table :analytics_dimension_dates
+  end
```

Comment:
You can just use `change`. It's smart enough to drop the table when rolled back.

```suggestion
  def change
    create_table :analytics_dimension_dates, id: false do |t|
      t.date :date, primary_key: true
      t.string :year
      t.string :month
      t.string :day
    end
  end
```

(as a side comment: if you need to implement the up-down explicitly, `reversible` is the preferred option these days.)